### PR TITLE
chore(svelte): Migrate all icons to Lucide/custom icons

### PR DESCRIPTION
**Pull request by fkling**
Closes srch-467

tl;dr: This commit switches all remaining mdi icons to lucide.

This commit does a couple of things to make the migration to lucide icons and to the `Icon2` (now `Icon` again) component complete:

- Use lucide icons instead of mdi icons
- Turn custom icons back to SVG files and use `unplugin-icon`s custom collections support to reference them
- Rename `Icon` component to `SVGIcon`. We still need it for integrating with shared web app code.
- Rename `Icon2` component to `Icon`.
- Update to svelte@4.2.18 to [bring in a fix for `<svelte:component`](https://github.com/sveltejs/svelte/releases/tag/svelte%404.2.12) which is used by (now) `Icon`.

I put our generic icons in `assets/icons/` and the specific symbol icons in `assets/symbol/icons`. They can be referenced via `ISgName` and `ISymbolName`. I'm open to changes to the location and the naming.

If the same icon wasn't available in lucide I tried to find one that made sense semantically. I expect we'll make another pass (in a separate PR) to adjust icons and tweak icon sizes/colors.

## Test plan

`pnpm build` and svelte check

### CODE REVIEW

File: client/web-sveltekit/src/lib/Icon.module.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@ $icon-inline-size: var(--icon-inline-size, #{(16 / 14)}em);
 svg.icon {
     width: $icon-size;
     height: $icon-size;
-    color: var(--icon-fill-color, var(--icon-color, inherit));
+    color: var(--icon-fill-color, var(--color, inherit));
```
Comment (fkling) [inline client/web-sveltekit/src/lib/Icon.module.scss:7]: This brings in line with how the original `Icon` component works. But we should probably rethink the API for colorization, I find it a bit confusing.

File: client/web-sveltekit/src/lib/Icon.svelte
Comment (fkling) [inline client/web-sveltekit/src/lib/Icon.svelte:1]: This was renamed from `Icon2`.

File: client/web-sveltekit/src/lib/SVGIcon.svelte
Comment (fkling) [inline client/web-sveltekit/src/lib/SVGIcon.svelte:1]: This was renamed from `Icon`.

File: client/web-sveltekit/src/lib/Icon.module.scss (hunk 1 above)
Comment (camdencheek) [reply client/web-sveltekit/src/lib/Icon.module.scss:7]: Ah, nice. I ran into this adding an icon in another PR I'm working on. Basically, `--icon-color` is set at the root, so `inherit` was never triggering. Solved it before I even got to ask about it 😄

File: client/web-sveltekit/src/lib/HeroPage.svelte [hunk 2]
```diff
@@ -1,15 +1,17 @@
 <script lang="ts">
+    import type { ComponentProps } from 'svelte'
+
     import Icon from '$lib/Icon.svelte'
 
     export let title: string
-    export let svgIconPath: string = ''
+    export let icon: ComponentProps<Icon>['icon']
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/HeroPage.svelte:7]: Oh, that's a _much_ nicer type name than what I came up with when I attempted this

Comment (camdencheek) [reply client/web-sveltekit/src/lib/HeroPage.svelte:7]: Q: why not use the exported `IconComponent` type from `Icon`?

File: client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte [hunk 3]
```diff
@@ -40,7 +40,12 @@
     {#if loading}
         <LoadingSpinner --size="16px" />
     {:else}
-        <Icon svgPath={icons[severity]} --icon-size="16px" --color={isError ? 'var(--danger)' : 'var(--text-title)'} />
+        <Icon
+            icon={icons[severity]}
+            aria-label={severity}
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte:45]: Thank you for auditing the aria properties while you're at it!

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPickerItem.svelte [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
+
     import Avatar from '$lib/Avatar.svelte'
     import Icon from '$lib/Icon.svelte'
     import Timestamp from '$lib/Timestamp.svelte'
+    import { Badge } from '$lib/wildcard'
+
     import type { RepositoryGitRevAuthor } from './RepositoryRevPicker.gql'
 
-    export let iconPath: string
+    export let icon: ComponentProps<Icon>['icon']|undefined = undefined
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/RepositoryRevPickerItem.svelte:11]: Similarly, why not use `IconComponent`?

**Review by camdencheek — APPROVED** [review]
Woot! Thanks for cleaning this up!

File: client/web-sveltekit/src/lib/HeroPage.svelte (hunk 2 above)
Comment (fkling) [reply client/web-sveltekit/src/lib/HeroPage.svelte:7]: I thought it makes to use this type in Svelte components where the Icon component is actually used, and use the dedicated type in TS files where we just define other types.

File: client/web-sveltekit/BUILD.bazel [hunk 5]
```diff
@@ -59,7 +59,6 @@ BUILD_DEPS = [
     "//:node_modules/@codemirror/search",
     "//:node_modules/@codemirror/state",
     "//:node_modules/@codemirror/view",
-    "//:node_modules/@mdi/js",
```
[the remark this answers is not in the document]
Comment (fkling) [reply client/web-sveltekit/BUILD.bazel:62]: I'll have to re-add it for the context specific suggestions 🤷‍♂️

File: client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte (hunk 3 above)
Comment (fkling) [reply client/web-sveltekit/src/lib/search/resultsIndicator/ResultsIndicator.svelte:45]: I could have done more but I didn't want to stuff too much into a single PR
